(window.webpackJsonp=window.webpackJsonp||[]).push([[18],{111:function(e,t,n){"use strict";n.d(t,"a",(function(){return u})),n.d(t,"b",(function(){return b}));var a=n(0),i=n.n(a);function o(e,t,n){return t in e?Object.defineProperty(e,t,{value:n,enumerable:!0,configurable:!0,writable:!0}):e[t]=n,e}function r(e,t){var n=Object.keys(e);if(Object.getOwnPropertySymbols){var a=Object.getOwnPropertySymbols(e);t&&(a=a.filter((function(t){return Object.getOwnPropertyDescriptor(e,t).enumerable}))),n.push.apply(n,a)}return n}function l(e){for(var t=1;t<arguments.length;t++){var n=null!=arguments[t]?arguments[t]:{};t%2?r(Object(n),!0).forEach((function(t){o(e,t,n[t])})):Object.getOwnPropertyDescriptors?Object.defineProperties(e,Object.getOwnPropertyDescriptors(n)):r(Object(n)).forEach((function(t){Object.defineProperty(e,t,Object.getOwnPropertyDescriptor(n,t))}))}return e}function s(e,t){if(null==e)return{};var n,a,i=function(e,t){if(null==e)return{};var n,a,i={},o=Object.keys(e);for(a=0;a<o.length;a++)n=o[a],t.indexOf(n)>=0||(i[n]=e[n]);return i}(e,t);if(Object.getOwnPropertySymbols){var o=Object.getOwnPropertySymbols(e);for(a=0;a<o.length;a++)n=o[a],t.indexOf(n)>=0||Object.prototype.propertyIsEnumerable.call(e,n)&&(i[n]=e[n])}return i}var d=i.a.createContext({}),c=function(e){var t=i.a.useContext(d),n=t;return e&&(n="function"==typeof e?e(t):l(l({},t),e)),n},u=function(e){var t=c(e.components);return i.a.createElement(d.Provider,{value:t},e.children)},m={inlineCode:"code",wrapper:function(e){var t=e.children;return i.a.createElement(i.a.Fragment,{},t)}},p=i.a.forwardRef((function(e,t){var n=e.components,a=e.mdxType,o=e.originalType,r=e.parentName,d=s(e,["components","mdxType","originalType","parentName"]),u=c(n),p=a,b=u["".concat(r,".").concat(p)]||u[p]||m[p]||o;return n?i.a.createElement(b,l(l({ref:t},d),{},{components:n})):i.a.createElement(b,l({ref:t},d))}));function b(e,t){var n=arguments,a=t&&t.mdxType;if("string"==typeof e||a){var o=n.length,r=new Array(o);r[0]=p;var l={};for(var s in t)hasOwnProperty.call(t,s)&&(l[s]=t[s]);l.originalType=e,l.mdxType="string"==typeof e?e:a,r[1]=l;for(var d=2;d<o;d++)r[d]=n[d];return i.a.createElement.apply(null,r)}return i.a.createElement.apply(null,n)}p.displayName="MDXCreateElement"},88:function(e,t,n){"use strict";n.r(t),n.d(t,"frontMatter",(function(){return r})),n.d(t,"metadata",(function(){return l})),n.d(t,"toc",(function(){return s})),n.d(t,"default",(function(){return c}));var a=n(3),i=n(7),o=(n(0),n(111)),r={last_modified_on:"2021-02-02",id:"harmonization",title:"Data harmonization"},l={unversionedId:"guides/harmonization",id:"guides/harmonization",isDocsHomePage:!1,title:"Data harmonization",description:"This step is necessary for testing and validation with an external dataset. You can skip this step if you don't have an external validation dataset.",source:"@site/docs/guides/harmonization.md",slug:"/guides/harmonization",permalink:"/docs/guides/harmonization",editUrl:"https://github.com/genoml/genoml.github.io/edit/main/docs/guides/harmonization.md",version:"current",sidebar:"docs",previous:{title:"Data munging",permalink:"/docs/guides/munging"},next:{title:"Model training",permalink:"/docs/guides/training"}},s=[{value:"Harmonizing your Validation/Test Dataset",id:"harmonizing-your-validationtest-dataset",children:[]}],d={toc:s};function c(e){var t=e.components,n=Object(i.a)(e,["components"]);return Object(o.b)("wrapper",Object(a.a)({},d,n,{components:t,mdxType:"MDXLayout"}),Object(o.b)("div",{className:"admonition admonition-info alert alert--info"},Object(o.b)("div",{parentName:"div",className:"admonition-heading"},Object(o.b)("h5",{parentName:"div"},Object(o.b)("span",{parentName:"h5",className:"admonition-icon"},Object(o.b)("svg",{parentName:"span",xmlns:"http://www.w3.org/2000/svg",width:"14",height:"16",viewBox:"0 0 14 16"},Object(o.b)("path",{parentName:"svg",fillRule:"evenodd",d:"M7 2.3c3.14 0 5.7 2.56 5.7 5.7s-2.56 5.7-5.7 5.7A5.71 5.71 0 0 1 1.3 8c0-3.14 2.56-5.7 5.7-5.7zM7 1C3.14 1 0 4.14 0 8s3.14 7 7 7 7-3.14 7-7-3.14-7-7-7zm1 3H6v5h2V4zm0 6H6v2h2v-2z"}))),"info")),Object(o.b)("div",{parentName:"div",className:"admonition-content"},Object(o.b)("p",{parentName:"div"},"This step is necessary for ",Object(o.b)("a",{parentName:"p",href:"/docs/guides/validation"},"testing and validation")," with an external dataset. You can skip this step if you don't have an external validation dataset. "))),Object(o.b)("p",null,"To properly test how your model performs on a dataset it's never seen before (but you start with different PLINK binaries), we have created the harmonization step that will:"),Object(o.b)("ol",null,Object(o.b)("li",{parentName:"ol"},"Keep only the same SNPs between your reference dataset and the dataset you are using for validation"),Object(o.b)("li",{parentName:"ol"},"Force the reference alleles in the validation dataset to match your reference dataset"),Object(o.b)("li",{parentName:"ol"},"Export a ",Object(o.b)("inlineCode",{parentName:"li"},".txt")," file with the column names from your reference dataset to later use in the munging of your validation dataset ")),Object(o.b)("p",null,"Using GenoML for both your reference dataset and then your validation dataset, the process will look like the following: "),Object(o.b)("ol",null,Object(o.b)("li",{parentName:"ol"},'Munge and train your first dataset. That will be your "reference" model'),Object(o.b)("li",{parentName:"ol"},"Use the outputs of step 1's munge for your reference model to harmonize your incoming validation dataset"),Object(o.b)("li",{parentName:"ol"},"Run through the harmonization step with your validation dataset"),Object(o.b)("li",{parentName:"ol"},"Run through munging with your newly harmonized dataset"),Object(o.b)("li",{parentName:"ol"},"Retrain your reference model with only the matching columns of your unseen data. Given the nature of ML algorithms, you cannot test a model on a set of data that does not have identical features"),Object(o.b)("li",{parentName:"ol"},"Test your newly retrained reference model on the unseen data")),Object(o.b)("div",{className:"admonition admonition-info alert alert--info"},Object(o.b)("div",{parentName:"div",className:"admonition-heading"},Object(o.b)("h5",{parentName:"div"},Object(o.b)("span",{parentName:"h5",className:"admonition-icon"},Object(o.b)("svg",{parentName:"span",xmlns:"http://www.w3.org/2000/svg",width:"14",height:"16",viewBox:"0 0 14 16"},Object(o.b)("path",{parentName:"svg",fillRule:"evenodd",d:"M7 2.3c3.14 0 5.7 2.56 5.7 5.7s-2.56 5.7-5.7 5.7A5.71 5.71 0 0 1 1.3 8c0-3.14 2.56-5.7 5.7-5.7zM7 1C3.14 1 0 4.14 0 8s3.14 7 7 7 7-3.14 7-7-3.14-7-7-7zm1 3H6v5h2V4zm0 6H6v2h2v-2z"}))),"info")),Object(o.b)("div",{parentName:"div",className:"admonition-content"},Object(o.b)("p",{parentName:"div"},"If you are using an external validation dataset, Steps 1-4 are performed here as part of the harmonization process. Steps 5-6 are performed as part of the ",Object(o.b)("a",{parentName:"p",href:"/docs/guides/validation"},"testing and validation")," process. "))),Object(o.b)("h3",{id:"harmonizing-your-validationtest-dataset"},"Harmonizing your Validation/Test Dataset"),Object(o.b)("p",null,Object(o.b)("strong",{parentName:"p"},"Required")," arguments for harmonizing with GenoML are the following: "),Object(o.b)("ul",null,Object(o.b)("li",{parentName:"ul"},Object(o.b)("inlineCode",{parentName:"li"},"--test_geno_prefix")," : What is the prefix of your validation dataset PLINK binaries?"),Object(o.b)("li",{parentName:"ul"},Object(o.b)("inlineCode",{parentName:"li"},"--test_prefix"),": What do you want the output to be named?"),Object(o.b)("li",{parentName:"ul"},Object(o.b)("inlineCode",{parentName:"li"},"--ref_model_prefix"),":  What is the name of the previously GenoML-munged dataset you would like to use as your reference dataset? (Without the ",Object(o.b)("inlineCode",{parentName:"li"},".dataForML.h5")," suffix)"),Object(o.b)("li",{parentName:"ul"},Object(o.b)("inlineCode",{parentName:"li"},"--training_snps_alleles")," : What are the SNPs and alleles you would like to use? (This is generated at the end of your previously-GenoML munged dataset with the suffix ",Object(o.b)("inlineCode",{parentName:"li"},"variants_and_alleles.tab"),")")),Object(o.b)("p",null,"To harmonize your incoming validation dataset to match the SNPs and alleles to your reference dataset, the command would look like the following:"),Object(o.b)("pre",null,Object(o.b)("code",{parentName:"pre",className:"language-shell"},"# Running GenoML harmonize\n\ngenoml harmonize \\\n--test_geno_prefix examples/discrete/validation \\\n--test_prefix outputs/validation_test_discrete_geno \\\n--ref_model_prefix outputs/test_discrete_geno \\\n--training_snps_alleles outputs/test_discrete_geno.variants_and_alleles.tab\n")),Object(o.b)("p",null,"This step will generate: "),Object(o.b)("ul",null,Object(o.b)("li",{parentName:"ul"},"a ",Object(o.b)("inlineCode",{parentName:"li"},"*_refColsHarmonize_toKeep.txt")," file of columns to keep for the next step "),Object(o.b)("li",{parentName:"ul"},Object(o.b)("inlineCode",{parentName:"li"},"*_refSNPs_andAlleles.*")," PLINK binary files (.bed, .bim, and .fam) that have the SNPs and alleles match your reference dataset")),Object(o.b)("p",null,"Now that you have harmonized your validation dataset to your reference dataset, you can now munge using a command similar to the following:"),Object(o.b)("pre",null,Object(o.b)("code",{parentName:"pre",className:"language-shell"},"# Running GenoML munge after GenoML harmonize\n\ngenoml discrete supervised munge --prefix outputs/validation_test_discrete_geno \\\n--geno outputs/validation_test_discrete_geno_refSNPs_andAlleles \\\n--pheno examples/discrete/validation_pheno.csv \\\n--addit examples/discrete/validation_addit.csv \\\n--ref_cols_harmonize outputs/validation_test_discrete_geno_refColsHarmonize_toKeep.txt\n")),Object(o.b)("p",null,"All munging options discussed above are available at this step; the only difference here is you will add the ",Object(o.b)("inlineCode",{parentName:"p"},"--ref_cols_harmonize")," flag to include the ",Object(o.b)("inlineCode",{parentName:"p"},"*.refColsHarmonize_toKeep.txt")," file generated at the end of harmonizing only to keep the same columns that the reference dataset had. "))}c.isMDXComponent=!0}}]);