(window.webpackJsonp=window.webpackJsonp||[]).push([[33],{102:function(e,n,t){"use strict";t.r(n),t.d(n,"frontMatter",(function(){return c})),t.d(n,"metadata",(function(){return i})),t.d(n,"toc",(function(){return s})),t.d(n,"default",(function(){return d}));var o=t(3),r=t(7),a=(t(0),t(110)),c={last_modified_on:"2021-03-02",id:"correcting-confounders",title:"Correcting confoundign"},i={unversionedId:"use-cases/correcting-confounders",id:"use-cases/correcting-confounders",isDocsHomePage:!1,title:"Correcting confoundign",description:"This section is under development.",source:"@site/docs/use-cases/correcting-confounders.md",slug:"/use-cases/correcting-confounders",permalink:"/docs/use-cases/correcting-confounders",editUrl:"https://github.com/genoml/genoml.github.io/edit/main/docs/use-cases/correcting-confounders.md",version:"current"},s=[{value:"Cohort stratification in GenoML",id:"cohort-stratification-in-genoml",children:[]}],u={toc:s};function d(e){var n=e.components,t=Object(r.a)(e,["components"]);return Object(a.b)("wrapper",Object(o.a)({},u,t,{components:n,mdxType:"MDXLayout"}),Object(a.b)("div",{className:"admonition admonition-note alert alert--secondary"},Object(a.b)("div",{parentName:"div",className:"admonition-heading"},Object(a.b)("h5",{parentName:"div"},Object(a.b)("span",{parentName:"h5",className:"admonition-icon"},Object(a.b)("svg",{parentName:"span",xmlns:"http://www.w3.org/2000/svg",width:"14",height:"16",viewBox:"0 0 14 16"},Object(a.b)("path",{parentName:"svg",fillRule:"evenodd",d:"M6.3 5.69a.942.942 0 0 1-.28-.7c0-.28.09-.52.28-.7.19-.18.42-.28.7-.28.28 0 .52.09.7.28.18.19.28.42.28.7 0 .28-.09.52-.28.7a1 1 0 0 1-.7.3c-.28 0-.52-.11-.7-.3zM8 7.99c-.02-.25-.11-.48-.31-.69-.2-.19-.42-.3-.69-.31H6c-.27.02-.48.13-.69.31-.2.2-.3.44-.31.69h1v3c.02.27.11.5.31.69.2.2.42.31.69.31h1c.27 0 .48-.11.69-.31.2-.19.3-.42.31-.69H8V7.98v.01zM7 2.3c-3.14 0-5.7 2.54-5.7 5.68 0 3.14 2.56 5.7 5.7 5.7s5.7-2.55 5.7-5.7c0-3.15-2.56-5.69-5.7-5.69v.01zM7 .98c3.86 0 7 3.14 7 7s-3.14 7-7 7-7-3.12-7-7 3.14-7 7-7z"}))),"note")),Object(a.b)("div",{parentName:"div",className:"admonition-content"},Object(a.b)("p",{parentName:"div"},"This section is under development."))),Object(a.b)("p",null,"Something cool that we have done in GenoML is to allow you at the ",Object(a.b)("inlineCode",{parentName:"p"},"munge"),"-ing phase is allow you to correct a dataset for potential confounding variables. This can be anything from correcting gene expresison data for study site and batch to correcting a genome-wide set of SNPs for population stratification.",Object(a.b)("br",{parentName:"p"}),"\n","Please check out the following options in the ",Object(a.b)("inlineCode",{parentName:"p"},"munge")," workflow to correct your data for know covariates. In future releases, we might help you correct for hidden or latent covariates as well."),Object(a.b)("pre",null,Object(a.b)("code",{parentName:"pre"},"  --umap_reduce {no,yes}\n                        Would you like to reduce your dimensions with UMAP?\n                        [default: no]. Must be run with --confounders flag if\n                        yes.\n  --adjust_data {yes,no}\n                        Would you like to adjust features and/or confounders\n                        in your data? [default: no]\n  --adjust_normalize {no,yes}\n                        Would you like to normalize the features and/or\n                        confounders you are adjusting for in your data?\n                        [default: yes]\n  --target_features TARGET_FEATURES\n                        For adjusting data. A .txt file, one column, with a\n                        list of features to adjust (no header). These should\n                        correspond to features in the munged dataset\n  --confounders CONFOUNDERS\n                        For adjusting data. A .csv of confounders to adjust\n                        for with ID column and header.Numeric, with no missing\n                        data and the ID columnis mandatory\n")),Object(a.b)("h2",{id:"cohort-stratification-in-genoml"},"Cohort stratification in GenoML"),Object(a.b)("p",null,"Run:"),Object(a.b)("pre",null,Object(a.b)("code",{parentName:"pre",className:"language-shell"},"#Todo\n")))}d.isMDXComponent=!0},110:function(e,n,t){"use strict";t.d(n,"a",(function(){return l})),t.d(n,"b",(function(){return m}));var o=t(0),r=t.n(o);function a(e,n,t){return n in e?Object.defineProperty(e,n,{value:t,enumerable:!0,configurable:!0,writable:!0}):e[n]=t,e}function c(e,n){var t=Object.keys(e);if(Object.getOwnPropertySymbols){var o=Object.getOwnPropertySymbols(e);n&&(o=o.filter((function(n){return Object.getOwnPropertyDescriptor(e,n).enumerable}))),t.push.apply(t,o)}return t}function i(e){for(var n=1;n<arguments.length;n++){var t=null!=arguments[n]?arguments[n]:{};n%2?c(Object(t),!0).forEach((function(n){a(e,n,t[n])})):Object.getOwnPropertyDescriptors?Object.defineProperties(e,Object.getOwnPropertyDescriptors(t)):c(Object(t)).forEach((function(n){Object.defineProperty(e,n,Object.getOwnPropertyDescriptor(t,n))}))}return e}function s(e,n){if(null==e)return{};var t,o,r=function(e,n){if(null==e)return{};var t,o,r={},a=Object.keys(e);for(o=0;o<a.length;o++)t=a[o],n.indexOf(t)>=0||(r[t]=e[t]);return r}(e,n);if(Object.getOwnPropertySymbols){var a=Object.getOwnPropertySymbols(e);for(o=0;o<a.length;o++)t=a[o],n.indexOf(t)>=0||Object.prototype.propertyIsEnumerable.call(e,t)&&(r[t]=e[t])}return r}var u=r.a.createContext({}),d=function(e){var n=r.a.useContext(u),t=n;return e&&(t="function"==typeof e?e(n):i(i({},n),e)),t},l=function(e){var n=d(e.components);return r.a.createElement(u.Provider,{value:n},e.children)},f={inlineCode:"code",wrapper:function(e){var n=e.children;return r.a.createElement(r.a.Fragment,{},n)}},p=r.a.forwardRef((function(e,n){var t=e.components,o=e.mdxType,a=e.originalType,c=e.parentName,u=s(e,["components","mdxType","originalType","parentName"]),l=d(t),p=o,m=l["".concat(c,".").concat(p)]||l[p]||f[p]||a;return t?r.a.createElement(m,i(i({ref:n},u),{},{components:t})):r.a.createElement(m,i({ref:n},u))}));function m(e,n){var t=arguments,o=n&&n.mdxType;if("string"==typeof e||o){var a=t.length,c=new Array(a);c[0]=p;var i={};for(var s in n)hasOwnProperty.call(n,s)&&(i[s]=n[s]);i.originalType=e,i.mdxType="string"==typeof e?e:o,c[1]=i;for(var u=2;u<a;u++)c[u]=t[u];return r.a.createElement.apply(null,c)}return r.a.createElement.apply(null,t)}p.displayName="MDXCreateElement"}}]);