<!doctype html>
<html lang="en">
<head>
<meta charset="UTF-8">
<meta name="viewport" content="width=device-width,initial-scale=1">
<meta name="generator" content="Docusaurus v2.0.0-alpha.70">
<link rel="alternate" type="application/rss+xml" href="/blog/rss.xml" title="GenoML Blog RSS Feed">
<link rel="alternate" type="application/atom+xml" href="/blog/atom.xml" title="GenoML Blog Atom Feed">
<link rel="preconnect" href="https://www.google-analytics.com">
<script>window.ga=window.ga||function(){(ga.q=ga.q||[]).push(arguments)},ga.l=+new Date,ga("create","UA-134657372-1","auto"),ga("send","pageview")</script>
<script async src="https://www.google-analytics.com/analytics.js"></script><title data-react-helmet="true">Testing and validation | GenoML</title><meta data-react-helmet="true" name="twitter:card" content="summary_large_image"><meta data-react-helmet="true" name="docusaurus_locale" content="en"><meta data-react-helmet="true" name="docusaurus_version" content="current"><meta data-react-helmet="true" name="docusaurus_tag" content="docs-default-current"><meta data-react-helmet="true" property="og:title" content="Testing and validation | GenoML"><meta data-react-helmet="true" name="description" content="To properly test how your model performs on a dataset it&#x27;s never seen before (but you start with different PLINK binaries), we have created the harmonization step that will:"><meta data-react-helmet="true" property="og:description" content="To properly test how your model performs on a dataset it&#x27;s never seen before (but you start with different PLINK binaries), we have created the harmonization step that will:"><meta data-react-helmet="true" property="og:url" content="https://genoml.com/docs/guides/validation"><link data-react-helmet="true" rel="shortcut icon" href="/img/favicon.ico"><link data-react-helmet="true" rel="canonical" href="https://genoml.com/docs/guides/validation"><link rel="stylesheet" href="/styles.29eb1201.css">
<link rel="preload" href="/styles.9c4bafb9.js" as="script">
<link rel="preload" href="/runtime~main.dc45b22d.js" as="script">
<link rel="preload" href="/main.8d3b9ef6.js" as="script">
<link rel="preload" href="/1.99cc7029.js" as="script">
<link rel="preload" href="/2.9d8f953f.js" as="script">
<link rel="preload" href="/37.2b52632f.js" as="script">
<link rel="preload" href="/38.8fd46ecf.js" as="script">
<link rel="preload" href="/935f2afb.5192d914.js" as="script">
<link rel="preload" href="/17896441.058f9384.js" as="script">
<link rel="preload" href="/9b438f6a.667c0b75.js" as="script">
</head>
<body>
<script>!function(){function t(t){document.documentElement.setAttribute("data-theme",t)}var e=function(){var t=null;try{t=localStorage.getItem("theme")}catch(t){}return t}();t(null!==e?e:"dark")}()</script><div id="__docusaurus">
<nav aria-label="Skip navigation links"><button type="button" tabindex="0" class="skipToContent_11B0">Skip to main content</button></nav><nav class="navbar navbar--fixed-top"><div class="navbar__inner"><div class="navbar__items"><div aria-label="Navigation bar toggle" class="navbar__toggle" role="button" tabindex="0"><svg aria-label="Menu" width="30" height="30" viewBox="0 0 30 30" role="img" focusable="false"><title>Menu</title><path stroke="currentColor" stroke-linecap="round" stroke-miterlimit="10" stroke-width="2" d="M4 7h22M4 15h22M4 23h22"></path></svg></div><a class="navbar__brand" href="/"><img src="/img/logo-cut.png" alt="GenoML Logo" class="themedImage_YANc themedImage--light_3CMI navbar__logo"><img src="/img/logo-cut.png" alt="GenoML Logo" class="themedImage_YANc themedImage--dark_3ARp navbar__logo"><strong class="navbar__title">GenoML</strong></a><a aria-current="page" class="navbar__item navbar__link navbar__link--active" href="/docs/">Docs</a><a class="navbar__item navbar__link" href="/blog/">Blog</a></div><div class="navbar__items navbar__items--right"><a href="https://github.com/genoml/genoml" target="_blank" rel="noopener noreferrer" class="navbar__item navbar__link">GitHub</a></div></div><div role="presentation" class="navbar-sidebar__backdrop"></div><div class="navbar-sidebar"><div class="navbar-sidebar__brand"><a class="navbar__brand" href="/"><img src="/img/logo-cut.png" alt="GenoML Logo" class="themedImage_YANc themedImage--light_3CMI navbar__logo"><img src="/img/logo-cut.png" alt="GenoML Logo" class="themedImage_YANc themedImage--dark_3ARp navbar__logo"><strong class="navbar__title">GenoML</strong></a></div><div class="navbar-sidebar__items"><div class="menu"><ul class="menu__list"><li class="menu__list-item"><a aria-current="page" class="menu__link navbar__link--active" href="/docs/">Docs</a></li><li class="menu__list-item"><a class="menu__link" href="/blog/">Blog</a></li><li class="menu__list-item"><a href="https://github.com/genoml/genoml" target="_blank" rel="noopener noreferrer" class="menu__link">GitHub</a></li></ul></div></div></div></nav><div class="main-wrapper"><div class="docPage_vMrn"><div class="docSidebarContainer_3Ak5" role="complementary"><div class="sidebar_3gvy"><div class="menu menu--responsive thin-scrollbar menu_1yIk"><button aria-label="Open Menu" aria-haspopup="true" class="button button--secondary button--sm menu__button" type="button"><svg aria-label="Menu" class="sidebarMenuIcon_1CUI" width="24" height="24" viewBox="0 0 30 30" role="img" focusable="false"><title>Menu</title><path stroke="currentColor" stroke-linecap="round" stroke-miterlimit="10" stroke-width="2" d="M4 7h22M4 15h22M4 23h22"></path></svg></button><ul class="menu__list"><li class="menu__list-item menu__list-item--collapsed"><a class="menu__link menu__link--sublist" href="#!">About</a><ul class="menu__list"><li class="menu__list-item"><a class="menu__link" tabindex="-1" href="/docs/about/what-is-genoml">What is GenoML?</a></li><li class="menu__list-item"><a class="menu__link" tabindex="-1" href="/docs/about/principles">Principles and philosophy</a></li><li class="menu__list-item"><a class="menu__link" tabindex="-1" href="/docs/about/roadmap">Roadmap</a></li><li class="menu__list-item"><a class="menu__link" tabindex="-1" href="/docs/about/contribute">Join and contribute</a></li></ul></li><li class="menu__list-item menu__list-item--collapsed"><a class="menu__link menu__link--sublist" href="#!">Setup</a><ul class="menu__list"><li class="menu__list-item"><a class="menu__link" tabindex="-1" href="/docs/setup/installation">Installation</a></li><li class="menu__list-item"><a class="menu__link" tabindex="-1" href="/docs/setup/configuration">Configuration</a></li></ul></li><li class="menu__list-item"><a class="menu__link menu__link--sublist menu__link--active" href="#!">Guides</a><ul class="menu__list"><li class="menu__list-item"><a class="menu__link" tabindex="0" href="/docs/guides/input">Input data</a></li><li class="menu__list-item"><a class="menu__link" tabindex="0" href="/docs/guides/munging">Data munging</a></li><li class="menu__list-item"><a class="menu__link" tabindex="0" href="/docs/guides/training">Model training</a></li><li class="menu__list-item"><a class="menu__link" tabindex="0" href="/docs/guides/tuning">Model tuning</a></li><li class="menu__list-item"><a aria-current="page" class="menu__link menu__link--active active" tabindex="0" href="/docs/guides/validation">Testing and validation</a></li><li class="menu__list-item"><a class="menu__link" tabindex="0" href="/docs/guides/experimental">Experimental features</a></li></ul></li><li class="menu__list-item menu__list-item--collapsed"><a class="menu__link menu__link--sublist" href="#!">Use cases</a><ul class="menu__list"><li class="menu__list-item"><a class="menu__link" tabindex="-1" href="/docs/use-cases/prs">Polygenic Risk Score (PRS)</a></li><li class="menu__list-item"><a class="menu__link" tabindex="-1" href="/docs/use-cases/target-identification">Target identification</a></li><li class="menu__list-item"><a class="menu__link" tabindex="-1" href="/docs/use-cases/cohort-stratification">Cohort stratification</a></li></ul></li><li class="menu__list-item menu__list-item--collapsed"><a class="menu__link menu__link--sublist" href="#!">Reference</a><ul class="menu__list"><li class="menu__list-item"><a class="menu__link" tabindex="-1" href="/docs/reference/cli">Command-Line Interface (CLI)</a></li></ul></li></ul></div></div></div><main class="docMainContainer_2iGs"><div class="container padding-vert--lg docItemWrapper_1bxp"><div class="row"><div class="col docItemCol_U38p"><div class="docItemContainer_a7m4"><article><header><h1 class="docTitle_Oumm">Testing and validation</h1></header><div class="markdown"><p>To properly test how your model performs on a dataset it&#x27;s never seen before (but you start with different PLINK binaries), we have created the harmonization step that will:</p><ol><li>Keep only the same SNPs between your reference dataset and the dataset you are using for validation</li><li>Force the reference alleles in the validation dataset to match your reference dataset</li><li>Export a <code>.txt</code> file with the column names from your reference dataset to later use in the munging of your validation dataset </li></ol><p>Using GenoML for both your reference dataset and then your validation dataset, the process will look like the following: </p><ol><li>Munge and train your first dataset<div class="mdxCodeBlock_1zKU"><div class="codeBlockContent_actS"><div tabindex="0" class="prism-code language-undefined codeBlock_tuNs thin-scrollbar"><div class="codeBlockLines_3uvA" style="color:#bfc7d5;background-color:#292d3e"><div class="token-line" style="color:#bfc7d5"><span class="token plain">- That will be your &quot;reference&quot; model</span></div></div></div><button type="button" aria-label="Copy code to clipboard" class="copyButton_2GIj">Copy</button></div></div></li><li>Use the outputs of step 1&#x27;s munge for your reference model to harmonize your incoming validation dataset</li><li>Run through the harmonization step with your validation dataset</li><li>Run through munging with your newly harmonized dataset</li><li>Retrain your reference model with only the matching columns of your unseen data <ul><li>Given the nature of ML algorithms, you cannot test a model on a set of data that does not have identical features</li></ul></li><li>Test your newly retrained reference model on the unseen data</li></ol><h3><a aria-hidden="true" tabindex="-1" class="anchor enhancedAnchor_prK2" id="harmonizing-your-validationtest-dataset"></a>Harmonizing your Validation/Test Dataset<a class="hash-link" href="#harmonizing-your-validationtest-dataset" title="Direct link to heading">#</a></h3><p><strong>Required</strong> arguments for harmonizing with GenoML are the following: </p><ul><li><code>--test_geno_prefix</code> : What is the prefix of your validation dataset PLINK binaries?</li><li><code>--test_prefix</code>: What do you want the output to be named?</li><li><code>--ref_model_prefix</code>:  What is the name of the previously GenoML-munged dataset you would like to use as your reference dataset? (Without the <code>.dataForML.h5</code> suffix)</li><li><code>--training_snps_alleles</code> : What are the SNPs and alleles you would like to use? (This is generated at the end of your previously-GenoML munged dataset with the suffix <code>variants_and_alleles.tab</code>)</li></ul><p>To harmonize your incoming validation dataset to match the SNPs and alleles to your reference dataset, the command would look like the following:</p><p>&quot;`shell</p><h1><a aria-hidden="true" tabindex="-1" class="anchor enhancedAnchor_prK2" id="running-genoml-harmonize"></a>Running GenoML harmonize<a class="hash-link" href="#running-genoml-harmonize" title="Direct link to heading">#</a></h1><p>genoml harmonize \
--test_geno_prefix examples/discrete/validation \
--test_prefix outputs/validation_test_discrete_geno \
--ref_model_prefix outputs/test_discrete_geno \
--training_snps_alleles outputs/test_discrete_geno.variants_and_alleles.tab</p><div class="mdxCodeBlock_1zKU"><div class="codeBlockContent_actS"><div tabindex="0" class="prism-code language-undefined codeBlock_tuNs thin-scrollbar"><div class="codeBlockLines_3uvA" style="color:#bfc7d5;background-color:#292d3e"><div class="token-line" style="color:#bfc7d5"><span class="token plain" style="display:inline-block">
</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">This step will generate: </span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">- a `*_refColsHarmonize_toKeep.txt` file of columns to keep for the next step </span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">- `*_refSNPs_andAlleles.*` PLINK binary files (.bed, .bim, and .fam) that have the SNPs and alleles match your reference dataset</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain" style="display:inline-block">
</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">Now that you have harmonized your validation dataset to your reference dataset, you can now munge using a command similar to the following:</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">&quot;`shell</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain"># Running GenoML munge after GenoML harmonize</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain" style="display:inline-block">
</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">genoml discrete supervised munge --prefix outputs/validation_test_discrete_geno \</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">--geno outputs/validation_test_discrete_geno_refSNPs_andAlleles \</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">--pheno examples/discrete/validation_pheno.csv \</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">--addit examples/discrete/validation_addit.csv \</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">--ref_cols_harmonize outputs/validation_test_discrete_geno_refColsHarmonize_toKeep.txt</span></div></div></div><button type="button" aria-label="Copy code to clipboard" class="copyButton_2GIj">Copy</button></div></div><p>All munging options discussed above are available at this step; the only difference here is you will add the <code>--ref_cols_harmonize</code> flag to include the <code>*.refColsHarmonize_toKeep.txt</code> file generated at the end of harmonizing only to keep the same columns that the reference dataset had. </p><p>After munging and training your reference model and harmonizing and munging your unseen test data, <strong>you will retrain your reference model to include only matching features</strong>. Given ML algorithms&#x27; nature, you cannot test a model on a set of data that does not have identical features. </p><p>To retrain your model appropriately, after munging your test data with the <code>--ref_cols_harmonize </code> flag, a final columns list will be generated at <code>*.finalHarmonizedCols_toKeep.txt</code>. This includes all the features that match between your unseen test data and your reference model. Use the <code>--matching_columns</code> flag when retraining your reference model to use the appropriate features.</p><p>When retraining of the reference model is complete, you are ready to test!</p><p>A step-by-step guide on how to achieve this is listed below:
&quot;`shell</p><h1><a aria-hidden="true" tabindex="-1" class="anchor enhancedAnchor_prK2" id="0-munge-the-reference-dataset"></a>0. MUNGE THE REFERENCE DATASET<a class="hash-link" href="#0-munge-the-reference-dataset" title="Direct link to heading">#</a></h1><p>genoml discrete supervised munge \
--prefix outputs/test_discrete_geno \
--geno examples/discrete/training \
--pheno examples/discrete/training_pheno.csv</p><h1><a aria-hidden="true" tabindex="-1" class="anchor enhancedAnchor_prK2" id="files-made"></a>Files made:<a class="hash-link" href="#files-made" title="Direct link to heading">#</a></h1><div class="mdxCodeBlock_1zKU"><div class="codeBlockContent_actS"><div tabindex="0" class="prism-code language-undefined codeBlock_tuNs thin-scrollbar"><div class="codeBlockLines_3uvA" style="color:#bfc7d5;background-color:#292d3e"><div class="token-line" style="color:#bfc7d5"><span class="token plain"># outputs/test_discrete_geno.dataForML.h5</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain"># outputs/test_discrete_geno.list_features.txt</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain"># outputs/test_discrete_geno.variants_and_alleles.tab</span></div></div></div><button type="button" aria-label="Copy code to clipboard" class="copyButton_2GIj">Copy</button></div></div><h1><a aria-hidden="true" tabindex="-1" class="anchor enhancedAnchor_prK2" id="1-train-the-reference-dataset"></a>1. TRAIN THE REFERENCE DATASET<a class="hash-link" href="#1-train-the-reference-dataset" title="Direct link to heading">#</a></h1><p>genoml discrete supervised train \
--prefix outputs/test_discrete_geno</p><h1><a aria-hidden="true" tabindex="-1" class="anchor enhancedAnchor_prK2" id="files-made-1"></a>Files made:<a class="hash-link" href="#files-made-1" title="Direct link to heading">#</a></h1><div class="mdxCodeBlock_1zKU"><div class="codeBlockContent_actS"><div tabindex="0" class="prism-code language-undefined codeBlock_tuNs thin-scrollbar"><div class="codeBlockLines_3uvA" style="color:#bfc7d5;background-color:#292d3e"><div class="token-line" style="color:#bfc7d5"><span class="token plain"># outputs/test_discrete_geno.best_algorithm.txt</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain"># outputs/test_discrete_geno.trainedModel.joblib</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain"># outputs/test_discrete_geno.trainedModel_trainingSample_Predictions.csv</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain"># outputs/test_discrete_geno.trainedModel_withheldSample_Predictions.csv</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain"># outputs/test_discrete_geno.trainedModel_withheldSample_ROC.png</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain"># outputs/test_discrete_geno.trainedModel_withheldSample_probabilities.png</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain"># outputs/test_discrete_geno.training_withheldSamples_performanceMetrics.csv</span></div></div></div><button type="button" aria-label="Copy code to clipboard" class="copyButton_2GIj">Copy</button></div></div><h1><a aria-hidden="true" tabindex="-1" class="anchor enhancedAnchor_prK2" id="2-harmonize-test-dataset-if-using-plinkgenotypes"></a>2. HARMONIZE TEST DATASET IF USING PLINK/GENOTYPES<a class="hash-link" href="#2-harmonize-test-dataset-if-using-plinkgenotypes" title="Direct link to heading">#</a></h1><p>genoml harmonize \
--test_geno_prefix examples/discrete/validation \
--test_prefix outputs/validation_test_discrete_geno \
--ref_model_prefix outputs/test_discrete_geno \
--training_snps_alleles outputs/test_discrete_geno.variants_and_alleles.tab</p><h1><a aria-hidden="true" tabindex="-1" class="anchor enhancedAnchor_prK2" id="files-made-2"></a>Files made:<a class="hash-link" href="#files-made-2" title="Direct link to heading">#</a></h1><div class="mdxCodeBlock_1zKU"><div class="codeBlockContent_actS"><div tabindex="0" class="prism-code language-undefined codeBlock_tuNs thin-scrollbar"><div class="codeBlockLines_3uvA" style="color:#bfc7d5;background-color:#292d3e"><div class="token-line" style="color:#bfc7d5"><span class="token plain"># outputs/validation_test_discrete_geno.refColsHarmonize_toKeep.txt</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain"># outputs/validation_test_discrete_geno.refSNPs_andAlleles.bed</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain"># outputs/validation_test_discrete_geno.refSNPs_andAlleles.bim</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain"># outputs/validation_test_discrete_geno.refSNPs_andAlleles.fam</span></div></div></div><button type="button" aria-label="Copy code to clipboard" class="copyButton_2GIj">Copy</button></div></div><h1><a aria-hidden="true" tabindex="-1" class="anchor enhancedAnchor_prK2" id="3-munge-the-test-dataset-on-reference-model-columns"></a>3. MUNGE THE TEST DATASET ON REFERENCE MODEL COLUMNS<a class="hash-link" href="#3-munge-the-test-dataset-on-reference-model-columns" title="Direct link to heading">#</a></h1><p>genoml discrete supervised munge \
--prefix outputs/validation_test_discrete_geno \
--geno outputs/validation_test_discrete_geno.refSNPs_andAlleles \
--pheno examples/discrete/validation_pheno.csv \
--addit examples/discrete/validation_addit.csv \
--ref_cols_harmonize outputs/validation_test_discrete_geno.refColsHarmonize_toKeep.txt</p><h1><a aria-hidden="true" tabindex="-1" class="anchor enhancedAnchor_prK2" id="files-made-3"></a>Files made:<a class="hash-link" href="#files-made-3" title="Direct link to heading">#</a></h1><div class="mdxCodeBlock_1zKU"><div class="codeBlockContent_actS"><div tabindex="0" class="prism-code language-undefined codeBlock_tuNs thin-scrollbar"><div class="codeBlockLines_3uvA" style="color:#bfc7d5;background-color:#292d3e"><div class="token-line" style="color:#bfc7d5"><span class="token plain"># outputs/validation_test_discrete_geno.finalHarmonizedCols_toKeep.txt</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain"># outputs/validation_test_discrete_geno.list_features.txt</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain"># outputs/test_discrete_geno.variants_and_alleles.tab</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain"># outputs/validation_test_discrete_geno.dataForML.h5</span></div></div></div><button type="button" aria-label="Copy code to clipboard" class="copyButton_2GIj">Copy</button></div></div><h1><a aria-hidden="true" tabindex="-1" class="anchor enhancedAnchor_prK2" id="4-retrain-reference-model-on-intersecting-columns-between-reference-and-test"></a>4. RETRAIN REFERENCE MODEL ON INTERSECTING COLUMNS BETWEEN REFERENCE AND TEST<a class="hash-link" href="#4-retrain-reference-model-on-intersecting-columns-between-reference-and-test" title="Direct link to heading">#</a></h1><p>genoml discrete supervised train \
--prefix outputs/test_discrete_geno \
--matching_columns outputs/validation_test_discrete_geno.finalHarmonizedCols_toKeep.txt</p><h1><a aria-hidden="true" tabindex="-1" class="anchor enhancedAnchor_prK2" id="note-this-replaces-the-trained-model-you-made-in-step-1"></a>Note: This replaces the trained model you made in step 1!<a class="hash-link" href="#note-this-replaces-the-trained-model-you-made-in-step-1" title="Direct link to heading">#</a></h1><h1><a aria-hidden="true" tabindex="-1" class="anchor enhancedAnchor_prK2" id="files-made-4"></a>Files made:<a class="hash-link" href="#files-made-4" title="Direct link to heading">#</a></h1><div class="mdxCodeBlock_1zKU"><div class="codeBlockContent_actS"><div tabindex="0" class="prism-code language-undefined codeBlock_tuNs thin-scrollbar"><div class="codeBlockLines_3uvA" style="color:#bfc7d5;background-color:#292d3e"><div class="token-line" style="color:#bfc7d5"><span class="token plain"># outputs/test_discrete_geno.best_algorithm.txt</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain"># outputs/test_discrete_geno.trainedModel.joblib</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain"># outputs/test_discrete_geno.trainedModel_trainingSample_Predictions.csv</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain"># outputs/test_discrete_geno.trainedModel_withheldSample_Predictions.csv</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain"># outputs/test_discrete_geno.trainedModel_withheldSample_ROC.png</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain"># outputs/test_discrete_geno.trainedModel_withheldSample_probabilities.png</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain"># outputs/test_discrete_geno.training_withheldSamples_performanceMetrics.csv</span></div></div></div><button type="button" aria-label="Copy code to clipboard" class="copyButton_2GIj">Copy</button></div></div><h1><a aria-hidden="true" tabindex="-1" class="anchor enhancedAnchor_prK2" id="5-test-retrained-reference-model-on-unseen-data"></a>5. TEST RETRAINED REFERENCE MODEL ON UNSEEN DATA<a class="hash-link" href="#5-test-retrained-reference-model-on-unseen-data" title="Direct link to heading">#</a></h1><p>genoml discrete supervised test \
--prefix outputs/validation_test_discrete_geno \
--test_prefix outputs/validation_test_discrete_geno \
--ref_model_prefix outputs/test_discrete_geno.trainedModel</p><h1><a aria-hidden="true" tabindex="-1" class="anchor enhancedAnchor_prK2" id="files-made-5"></a>Files made:<a class="hash-link" href="#files-made-5" title="Direct link to heading">#</a></h1><div class="mdxCodeBlock_1zKU"><div class="codeBlockContent_actS"><div tabindex="0" class="prism-code language-undefined codeBlock_tuNs thin-scrollbar"><div class="codeBlockLines_3uvA" style="color:#bfc7d5;background-color:#292d3e"><div class="token-line" style="color:#bfc7d5"><span class="token plain"># outputs/validation_test_discrete_geno.testedModel_allSample_predictions.csv</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain"># outputs/validation_test_discrete_geno.testedModel_allSample_probabilities.png</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain"># outputs/validation_test_discrete_geno.testedModel_allSample_ROC.png</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain"># outputs/validation_test_discrete_geno.testedModel_allSamples_performanceMetrics.csv</span></div></div></div><button type="button" aria-label="Copy code to clipboard" class="copyButton_2GIj">Copy</button></div></div><div class="mdxCodeBlock_1zKU"><div class="codeBlockContent_actS"><div tabindex="0" class="prism-code language-undefined codeBlock_tuNs thin-scrollbar"><div class="codeBlockLines_3uvA" style="color:#bfc7d5;background-color:#292d3e"><div class="token-line" style="color:#bfc7d5"><span class="token plain" style="display:inline-block">
</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">&gt; *Note:* When munging the test dataset on the reference model columns using the --ref_cols_harmonize, be sure not to include the --feature_selection flag, as you have already specified the columns to keep moving forward.</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain" style="display:inline-block">
</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">&lt;a id=&quot;5&quot;&gt;&lt;/a&gt;</span></div></div></div><button type="button" aria-label="Copy code to clipboard" class="copyButton_2GIj">Copy</button></div></div></div></article><div class="margin-vert--xl"><div class="row"><div class="col"><a href="https://github.com/genoml/genoml.github.io/edit/main/docs/guides/validation.md" target="_blank" rel="noreferrer noopener"><svg fill="currentColor" height="1.2em" width="1.2em" preserveAspectRatio="xMidYMid meet" role="img" viewBox="0 0 40 40" class="iconEdit_2LL7"><g><path d="m34.5 11.7l-3 3.1-6.3-6.3 3.1-3q0.5-0.5 1.2-0.5t1.1 0.5l3.9 3.9q0.5 0.4 0.5 1.1t-0.5 1.2z m-29.5 17.1l18.4-18.5 6.3 6.3-18.4 18.4h-6.3v-6.2z"></path></g></svg>Edit this page</a></div></div></div><div class="margin-vert--lg"><nav class="pagination-nav" aria-label="Blog list page navigation"><div class="pagination-nav__item"><a class="pagination-nav__link" href="/docs/guides/tuning"><div class="pagination-nav__sublabel">Previous</div><div class="pagination-nav__label">« Model tuning</div></a></div><div class="pagination-nav__item pagination-nav__item--next"><a class="pagination-nav__link" href="/docs/guides/experimental"><div class="pagination-nav__sublabel">Next</div><div class="pagination-nav__label">Experimental features »</div></a></div></nav></div></div></div><div class="col col--3"><div class="tableOfContents_2xL- thin-scrollbar"><ul class="table-of-contents table-of-contents__left-border"><li><a href="#harmonizing-your-validationtest-dataset" class="table-of-contents__link">Harmonizing your Validation/Test Dataset</a></li></ul></div></div></div></div></main></div></div><footer class="footer footer--dark"><div class="container"><div class="row footer__links"><div class="col footer__col"><h4 class="footer__title">Docs</h4><ul class="footer__items"><li class="footer__item"><a class="footer__link-item" href="/docs/about/what-is-genoml/">Getting Started</a></li><li class="footer__item"><a class="footer__link-item" href="/docs/about/contribute">Contribute</a></li></ul></div><div class="col footer__col"><h4 class="footer__title">Community</h4><ul class="footer__items"><li class="footer__item"><a href="https://stackoverflow.com/questions/tagged/genoml" target="_blank" rel="noopener noreferrer" class="footer__link-item">Stack Overflow</a></li><li class="footer__item"><a href="https://twitter.com/geno_ml" target="_blank" rel="noopener noreferrer" class="footer__link-item">Twitter</a></li><li class="footer__item"><a href="https://discordapp.com/invite/genoml" target="_blank" rel="noopener noreferrer" class="footer__link-item">Discord</a></li></ul></div><div class="col footer__col"><h4 class="footer__title">More</h4><ul class="footer__items"><li class="footer__item"><a class="footer__link-item" href="/blog">Blog</a></li><li class="footer__item"><a href="https://github.com/genoml/genoml" target="_blank" rel="noopener noreferrer" class="footer__link-item">GitHub</a></li></ul></div></div><div class="footer__bottom text--center"><div class="footer__copyright">Copyright © 2021 Data Tecnica International, LLC.</div></div></div></footer></div>
<script src="/styles.9c4bafb9.js"></script>
<script src="/runtime~main.dc45b22d.js"></script>
<script src="/main.8d3b9ef6.js"></script>
<script src="/1.99cc7029.js"></script>
<script src="/2.9d8f953f.js"></script>
<script src="/37.2b52632f.js"></script>
<script src="/38.8fd46ecf.js"></script>
<script src="/935f2afb.5192d914.js"></script>
<script src="/17896441.058f9384.js"></script>
<script src="/9b438f6a.667c0b75.js"></script>
</body>
</html>