<!doctype html>
<html lang="en">
<head>
<meta charset="UTF-8">
<meta name="viewport" content="width=device-width,initial-scale=1">
<meta name="generator" content="Docusaurus v2.0.0-alpha.70">
<link rel="alternate" type="application/rss+xml" href="/blog/rss.xml" title="GenoML Blog RSS Feed">
<link rel="alternate" type="application/atom+xml" href="/blog/atom.xml" title="GenoML Blog Atom Feed">
<link rel="preconnect" href="https://www.google-analytics.com">
<script>window.ga=window.ga||function(){(ga.q=ga.q||[]).push(arguments)},ga.l=+new Date,ga("create","UA-134657372-1","auto"),ga("send","pageview")</script>
<script async src="https://www.google-analytics.com/analytics.js"></script><title data-react-helmet="true">Command-Line Interface (CLI) | GenoML</title><meta data-react-helmet="true" name="twitter:card" content="summary_large_image"><meta data-react-helmet="true" name="docusaurus_locale" content="en"><meta data-react-helmet="true" name="docusaurus_version" content="current"><meta data-react-helmet="true" name="docusaurus_tag" content="docs-default-current"><meta data-react-helmet="true" property="og:title" content="Command-Line Interface (CLI) | GenoML"><meta data-react-helmet="true" name="description" content="If you would like to work with a continuous or discrete outcome in your \*.pheno file with genoml continuous or genoml discrete."><meta data-react-helmet="true" property="og:description" content="If you would like to work with a continuous or discrete outcome in your \*.pheno file with genoml continuous or genoml discrete."><meta data-react-helmet="true" property="og:url" content="https://genoml.com/docs/reference/cli"><link data-react-helmet="true" rel="shortcut icon" href="/img/favicon.ico"><link data-react-helmet="true" rel="canonical" href="https://genoml.com/docs/reference/cli"><link rel="stylesheet" href="/styles.29eb1201.css">
<link rel="preload" href="/styles.9c4bafb9.js" as="script">
<link rel="preload" href="/runtime~main.dc45b22d.js" as="script">
<link rel="preload" href="/main.8d3b9ef6.js" as="script">
<link rel="preload" href="/1.99cc7029.js" as="script">
<link rel="preload" href="/2.9d8f953f.js" as="script">
<link rel="preload" href="/37.2b52632f.js" as="script">
<link rel="preload" href="/38.8fd46ecf.js" as="script">
<link rel="preload" href="/935f2afb.5192d914.js" as="script">
<link rel="preload" href="/17896441.058f9384.js" as="script">
<link rel="preload" href="/91c76d4c.eb124e00.js" as="script">
</head>
<body>
<script>!function(){function t(t){document.documentElement.setAttribute("data-theme",t)}var e=function(){var t=null;try{t=localStorage.getItem("theme")}catch(t){}return t}();t(null!==e?e:"dark")}()</script><div id="__docusaurus">
<nav aria-label="Skip navigation links"><button type="button" tabindex="0" class="skipToContent_11B0">Skip to main content</button></nav><nav class="navbar navbar--fixed-top"><div class="navbar__inner"><div class="navbar__items"><div aria-label="Navigation bar toggle" class="navbar__toggle" role="button" tabindex="0"><svg aria-label="Menu" width="30" height="30" viewBox="0 0 30 30" role="img" focusable="false"><title>Menu</title><path stroke="currentColor" stroke-linecap="round" stroke-miterlimit="10" stroke-width="2" d="M4 7h22M4 15h22M4 23h22"></path></svg></div><a class="navbar__brand" href="/"><img src="/img/logo-cut.png" alt="GenoML Logo" class="themedImage_YANc themedImage--light_3CMI navbar__logo"><img src="/img/logo-cut.png" alt="GenoML Logo" class="themedImage_YANc themedImage--dark_3ARp navbar__logo"><strong class="navbar__title">GenoML</strong></a><a aria-current="page" class="navbar__item navbar__link navbar__link--active" href="/docs/">Docs</a><a class="navbar__item navbar__link" href="/blog/">Blog</a></div><div class="navbar__items navbar__items--right"><a href="https://github.com/genoml/genoml" target="_blank" rel="noopener noreferrer" class="navbar__item navbar__link">GitHub</a></div></div><div role="presentation" class="navbar-sidebar__backdrop"></div><div class="navbar-sidebar"><div class="navbar-sidebar__brand"><a class="navbar__brand" href="/"><img src="/img/logo-cut.png" alt="GenoML Logo" class="themedImage_YANc themedImage--light_3CMI navbar__logo"><img src="/img/logo-cut.png" alt="GenoML Logo" class="themedImage_YANc themedImage--dark_3ARp navbar__logo"><strong class="navbar__title">GenoML</strong></a></div><div class="navbar-sidebar__items"><div class="menu"><ul class="menu__list"><li class="menu__list-item"><a aria-current="page" class="menu__link navbar__link--active" href="/docs/">Docs</a></li><li class="menu__list-item"><a class="menu__link" href="/blog/">Blog</a></li><li class="menu__list-item"><a href="https://github.com/genoml/genoml" target="_blank" rel="noopener noreferrer" class="menu__link">GitHub</a></li></ul></div></div></div></nav><div class="main-wrapper"><div class="docPage_vMrn"><div class="docSidebarContainer_3Ak5" role="complementary"><div class="sidebar_3gvy"><div class="menu menu--responsive thin-scrollbar menu_1yIk"><button aria-label="Open Menu" aria-haspopup="true" class="button button--secondary button--sm menu__button" type="button"><svg aria-label="Menu" class="sidebarMenuIcon_1CUI" width="24" height="24" viewBox="0 0 30 30" role="img" focusable="false"><title>Menu</title><path stroke="currentColor" stroke-linecap="round" stroke-miterlimit="10" stroke-width="2" d="M4 7h22M4 15h22M4 23h22"></path></svg></button><ul class="menu__list"><li class="menu__list-item menu__list-item--collapsed"><a class="menu__link menu__link--sublist" href="#!">About</a><ul class="menu__list"><li class="menu__list-item"><a class="menu__link" tabindex="-1" href="/docs/about/what-is-genoml">What is GenoML?</a></li><li class="menu__list-item"><a class="menu__link" tabindex="-1" href="/docs/about/principles">Principles and philosophy</a></li><li class="menu__list-item"><a class="menu__link" tabindex="-1" href="/docs/about/roadmap">Roadmap</a></li><li class="menu__list-item"><a class="menu__link" tabindex="-1" href="/docs/about/contribute">Join and contribute</a></li></ul></li><li class="menu__list-item menu__list-item--collapsed"><a class="menu__link menu__link--sublist" href="#!">Setup</a><ul class="menu__list"><li class="menu__list-item"><a class="menu__link" tabindex="-1" href="/docs/setup/installation">Installation</a></li><li class="menu__list-item"><a class="menu__link" tabindex="-1" href="/docs/setup/configuration">Configuration</a></li></ul></li><li class="menu__list-item menu__list-item--collapsed"><a class="menu__link menu__link--sublist" href="#!">Guides</a><ul class="menu__list"><li class="menu__list-item"><a class="menu__link" tabindex="-1" href="/docs/guides/input">Input data</a></li><li class="menu__list-item"><a class="menu__link" tabindex="-1" href="/docs/guides/munging">Data munging</a></li><li class="menu__list-item"><a class="menu__link" tabindex="-1" href="/docs/guides/training">Model training</a></li><li class="menu__list-item"><a class="menu__link" tabindex="-1" href="/docs/guides/tuning">Model tuning</a></li><li class="menu__list-item"><a class="menu__link" tabindex="-1" href="/docs/guides/validation">Testing and validation</a></li><li class="menu__list-item"><a class="menu__link" tabindex="-1" href="/docs/guides/experimental">Experimental features</a></li></ul></li><li class="menu__list-item menu__list-item--collapsed"><a class="menu__link menu__link--sublist" href="#!">Use cases</a><ul class="menu__list"><li class="menu__list-item"><a class="menu__link" tabindex="-1" href="/docs/use-cases/prs">Polygenic Risk Score (PRS)</a></li><li class="menu__list-item"><a class="menu__link" tabindex="-1" href="/docs/use-cases/target-identification">Target identification</a></li><li class="menu__list-item"><a class="menu__link" tabindex="-1" href="/docs/use-cases/cohort-stratification">Cohort stratification</a></li></ul></li><li class="menu__list-item"><a class="menu__link menu__link--sublist menu__link--active" href="#!">Reference</a><ul class="menu__list"><li class="menu__list-item"><a aria-current="page" class="menu__link menu__link--active active" tabindex="0" href="/docs/reference/cli">Command-Line Interface (CLI)</a></li></ul></li></ul></div></div></div><main class="docMainContainer_2iGs"><div class="container padding-vert--lg docItemWrapper_1bxp"><div class="row"><div class="col docItemCol_U38p"><div class="docItemContainer_a7m4"><article><header><h1 class="docTitle_Oumm">Command-Line Interface (CLI)</h1></header><div class="markdown"><p>If you would like to work with a continuous or discrete outcome in your *.pheno file with <code>genoml continuous</code> or <code>genoml discrete</code>.  </p><p>The current iteration of documentation supports the <code>supervised</code> learning workflow; the list of workflows incoming is scary!
After that, you can add one of the subcommands, including choice of <code>munge</code>, <code>train</code>, <code>tune</code>, or <code>test</code>, these are steps within the supervised workflow.  </p><p>If you are interested in harmonizing a test dataset for external validation of a previous model, use <code>genoml harmonize</code>.</p><p>The general structure of GenoML commands is <code>genoml</code> then <code>outcome</code> type (discrete or continuous for now), then <code>workflow</code> (supervised for now), then <code>subcommand</code> that is usually a step within a larger workflow.</p><p>Detailed command line options relating to published GenoML <code>workflows</code> and their <code>subcommands</code> below. </p><h2><a aria-hidden="true" tabindex="-1" class="anchor enhancedAnchor_prK2" id="genoml---the-root-of-all-commands"></a>genoml - the root of all commands<a class="hash-link" href="#genoml---the-root-of-all-commands" title="Direct link to heading">#</a></h2><div class="mdxCodeBlock_1zKU"><div class="codeBlockContent_actS"><div tabindex="0" class="prism-code language-undefined codeBlock_tuNs thin-scrollbar"><div class="codeBlockLines_3uvA" style="color:#bfc7d5;background-color:#292d3e"><div class="token-line" style="color:#bfc7d5"><span class="token plain">usage: genoml &lt;command&gt; [&lt;args&gt;]</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">   continuous      for processing continuous datatypes (ex: age at onset)</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">   discrete        for processing discrete datatypes (ex: case vs. control status)</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">   harmonize       for harmonizing incoming test datasets to use the same SNPs and reference alleles prior to munging, training, and testing</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain" style="display:inline-block">
</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">genoml</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain" style="display:inline-block">
</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">positional arguments:</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">  command     Subcommand to run</span></div></div></div><button type="button" aria-label="Copy code to clipboard" class="copyButton_2GIj">Copy</button></div></div><h2><a aria-hidden="true" tabindex="-1" class="anchor enhancedAnchor_prK2" id="genoml-harmonize"></a>genoml harmonize<a class="hash-link" href="#genoml-harmonize" title="Direct link to heading">#</a></h2><div class="mdxCodeBlock_1zKU"><div class="codeBlockContent_actS"><div tabindex="0" class="prism-code language-undefined codeBlock_tuNs thin-scrollbar"><div class="codeBlockLines_3uvA" style="color:#bfc7d5;background-color:#292d3e"><div class="token-line" style="color:#bfc7d5"><span class="token plain">usage: genoml harmonize [-h] --test_geno_prefix TEST_GENO_PREFIX</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">                        [--test_prefix TEST_PREFIX]</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">                        [--ref_model_prefix REF_MODEL_PREFIX]</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">                        --training_snps_alleles TRAINING_SNPS_ALLELES [-v]</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain" style="display:inline-block">
</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">optional arguments:</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">  -h, --help            show this help message and exit</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">  --test_geno_prefix TEST_GENO_PREFIX</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">                        Prefix of the genotypes for the test dataset in PLINK</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">                        binary format.</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">  --test_prefix TEST_PREFIX</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">                        Prefix for the dataset you would like to test against</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">                        your reference model. Remember, the model will not</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">                        function well if it does not include the same</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">                        features, and these features should be on the same</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">                        numeric scale, you can leave off the &#x27;.dataForML.h5&#x27;</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">                        suffix.</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">  --ref_model_prefix REF_MODEL_PREFIX</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">                        Prefix of your reference model file, you can leave off</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">                        the &#x27;.joblib&#x27; suffix.</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">  --training_snps_alleles TRAINING_SNPS_ALLELES</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">                        File to the SNPs and alleles file generated in the</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">                        training phase that we will use to compare.</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">  -v, --verbose         Verbose output.</span></div></div></div><button type="button" aria-label="Copy code to clipboard" class="copyButton_2GIj">Copy</button></div></div><h2><a aria-hidden="true" tabindex="-1" class="anchor enhancedAnchor_prK2" id="genoml-discrete-supervised-munge"></a>genoml discrete supervised munge<a class="hash-link" href="#genoml-discrete-supervised-munge" title="Direct link to heading">#</a></h2><div class="mdxCodeBlock_1zKU"><div class="codeBlockContent_actS"><div tabindex="0" class="prism-code language-undefined codeBlock_tuNs thin-scrollbar"><div class="codeBlockLines_3uvA" style="color:#bfc7d5;background-color:#292d3e"><div class="token-line" style="color:#bfc7d5"><span class="token plain">usage: genoml discrete supervised munge [-h] [--prefix PREFIX]</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">                                        [--impute {median,mean}] [--geno GENO]</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">                                        --pheno PHENO [--addit ADDIT]</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">                                        [--feature_selection FEATURE_SELECTION]</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">                                        [--gwas GWAS] [--p P] [--vif VIF]</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">                                        [--iter ITER]</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">                                        [--ref_cols_harmonize REF_COLS_HARMONIZE]</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">                                        [-v]</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain" style="display:inline-block">
</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">optional arguments:</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">  -h, --help            show this help message and exit</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">  --prefix PREFIX       Prefix for your output build.</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">  --impute {median,mean}</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">                        Imputation: (mean, median). Governs secondary</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">                        imputation and data transformation [default: median].</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">  --geno GENO           Genotype: (string file path). Path to PLINK format</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">                        genotype file, everything before the *.bed/bim/fam</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">                        [default: None].</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">  --pheno PHENO         Phenotype: (string file path). Path to CSV phenotype</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">                        file [default: lost].</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">  --addit ADDIT         Additional: (string file path). Path to CSV format</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">                        feature file [default: None].</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">  --feature_selection FEATURE_SELECTION</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">                        Run a quick tree-based feature selection routine prior</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">                        to anything else, here you input the integer number of</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">                        estimators needed, we suggest &gt;= 50. The default of 0</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">                        will skip this functionality. This will also output a</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">                        reduced dataset for analyses in addition to feature</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">                        ranks. [default: 0]</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">  --gwas GWAS           GWAS summary stats: (string file path). Path to CSV</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">                        format external GWAS summary statistics containing at</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">                        least the columns SNP and P in the header [default:</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">                        nope].</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">  --p P                 P threshold for GWAS: (some value between 0-1). P</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">                        value to filter your SNP data on [default: 0.001].</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">  --vif VIF             Variance Inflation Factor (VIF): (integer). This is</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">                        the VIF threshold for pruning non-genotype features.</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">                        We recommend a value of 5-10. The default of 0 means</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">                        no VIF filtering will be done. [default: 0].</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">  --iter ITER           Iterator: (integer). How many iterations of VIF</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">                        pruning of features do you want to run. To save time</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">                        VIF is run in randomly assorted chunks of 1000</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">                        features per iteration. The default of 1 means only</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">                        one pass through the data. [default: 1].</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">  --ref_cols_harmonize REF_COLS_HARMONIZE</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">                        Are you now munging a test dataset following the</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">                        harmonize step? Here you input the path to the to the</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">                        *_refColsHarmonize_toKeep.txt file generated at that</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">                        step.</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">  -v, --verbose         Verbose output.</span></div></div></div><button type="button" aria-label="Copy code to clipboard" class="copyButton_2GIj">Copy</button></div></div><h2><a aria-hidden="true" tabindex="-1" class="anchor enhancedAnchor_prK2" id="genoml-continuous-supervised-munge"></a>genoml continuous supervised munge<a class="hash-link" href="#genoml-continuous-supervised-munge" title="Direct link to heading">#</a></h2><div class="mdxCodeBlock_1zKU"><div class="codeBlockContent_actS"><div tabindex="0" class="prism-code language-undefined codeBlock_tuNs thin-scrollbar"><div class="codeBlockLines_3uvA" style="color:#bfc7d5;background-color:#292d3e"><div class="token-line" style="color:#bfc7d5"><span class="token plain">usage: genoml continuous supervised munge [-h] [--prefix PREFIX]</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">                                          [--impute {median,mean}]</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">                                          [--geno GENO] --pheno PHENO</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">                                          [--addit ADDIT]</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">                                          [--feature_selection FEATURE_SELECTION]</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">                                          [--gwas GWAS] [--p P] [--vif VIF]</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">                                          [--iter ITER]</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">                                          [--ref_cols_harmonize REF_COLS_HARMONIZE]</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">                                          [-v]</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain" style="display:inline-block">
</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">optional arguments:</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">  -h, --help            show this help message and exit</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">  --prefix PREFIX       Prefix for your output build.</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">  --impute {median,mean}</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">                        Imputation: (mean, median). Governs secondary</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">                        imputation and data transformation [default: median].</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">  --geno GENO           Genotype: (string file path). Path to PLINK format</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">                        genotype file, everything before the *.bed/bim/fam</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">                        [default: None].</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">  --pheno PHENO         Phenotype: (string file path). Path to CSV phenotype</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">                        file [default: lost].</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">  --addit ADDIT         Additional: (string file path). Path to CSV format</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">                        feature file [default: None].</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">  --feature_selection FEATURE_SELECTION</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">                        Run a quick tree-based feature selection routine prior</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">                        to anything else, here you input the integer number of</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">                        estimators needed, we suggest &gt;= 50. The default of 0</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">                        will skip this functionality. This will also output a</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">                        reduced dataset for analyses in addition to feature</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">                        ranks. [default: 0]</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">  --gwas GWAS           GWAS summary stats: (string file path). Path to CSV</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">                        format external GWAS summary statistics containing at</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">                        least the columns SNP and P in the header [default:</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">                        nope].</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">  --p P                 P threshold for GWAS: (some value between 0-1). P</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">                        value to filter your SNP data on [default: 0.001].</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">  --vif VIF             Variance Inflation Factor (VIF): (integer). This is</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">                        the VIF threshold for pruning non-genotype features.</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">                        We recommend a value of 5-10. The default of 0 means</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">                        no VIF filtering will be done. [default: 0].</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">  --iter ITER           Iterator: (integer). How many iterations of VIF</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">                        pruning of features do you want to run. To save time</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">                        VIF is run in randomly assorted chunks of 1000</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">                        features per iteration. The default of 1 means only</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">                        one pass through the data. [default: 1].</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">  --ref_cols_harmonize REF_COLS_HARMONIZE</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">                        Are you now munging a test dataset following the</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">                        harmonize step? Here you input the path to the to the</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">                        *_refColsHarmonize_toKeep.txt file generated at that</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">                        step.</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">  -v, --verbose         Verbose output.</span></div></div></div><button type="button" aria-label="Copy code to clipboard" class="copyButton_2GIj">Copy</button></div></div><h2><a aria-hidden="true" tabindex="-1" class="anchor enhancedAnchor_prK2" id="genoml-discrete-supervised-train"></a>genoml discrete supervised train<a class="hash-link" href="#genoml-discrete-supervised-train" title="Direct link to heading">#</a></h2><div class="mdxCodeBlock_1zKU"><div class="codeBlockContent_actS"><div tabindex="0" class="prism-code language-undefined codeBlock_tuNs thin-scrollbar"><div class="codeBlockLines_3uvA" style="color:#bfc7d5;background-color:#292d3e"><div class="token-line" style="color:#bfc7d5"><span class="token plain">usage: genoml discrete supervised train [-h] [--prefix PREFIX]</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">                                        [--metric_max {AUC,Balanced_Accuracy,Specificity,Sensitivity}]</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">                                        [--prob_hist PROB_HIST] [--auc AUC]</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">                                        [--matching_columns MATCHING_COLUMNS]</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">                                        [-v]</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain" style="display:inline-block">
</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">optional arguments:</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">  -h, --help            show this help message and exit</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">  --prefix PREFIX       Prefix for your output build.</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">  --metric_max {AUC,Balanced_Accuracy,Specificity,Sensitivity}</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">                        How do you want to determine which algorithm performed</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">                        the best? [default: AUC].</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">  --prob_hist PROB_HIST</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">  --auc AUC</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">  --matching_columns MATCHING_COLUMNS</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">                        This is the list of intersecting columns between</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">                        reference and testing datasets with the suffix</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">                        *_finalHarmonizedCols_toKeep.txt</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">  -v, --verbose         Verbose output.</span></div></div></div><button type="button" aria-label="Copy code to clipboard" class="copyButton_2GIj">Copy</button></div></div><h2><a aria-hidden="true" tabindex="-1" class="anchor enhancedAnchor_prK2" id="genoml-continuous-supervised-train"></a>genoml continuous supervised train<a class="hash-link" href="#genoml-continuous-supervised-train" title="Direct link to heading">#</a></h2><div class="mdxCodeBlock_1zKU"><div class="codeBlockContent_actS"><div tabindex="0" class="prism-code language-undefined codeBlock_tuNs thin-scrollbar"><div class="codeBlockLines_3uvA" style="color:#bfc7d5;background-color:#292d3e"><div class="token-line" style="color:#bfc7d5"><span class="token plain">usage: genoml continuous supervised train [-h] [--prefix PREFIX]</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">                                          [--export_predictions EXPORT_PREDICTIONS]</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">                                          [--matching_columns MATCHING_COLUMNS]</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">                                          [-v]</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain" style="display:inline-block">
</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">optional arguments:</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">  -h, --help            show this help message and exit</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">  --prefix PREFIX       Prefix for your output build.</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">  --export_predictions EXPORT_PREDICTIONS</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">  --matching_columns MATCHING_COLUMNS</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">                        This is the list of intersecting columns between</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">                        reference and testing datasets with the suffix</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">                        *_finalHarmonizedCols_toKeep.txt</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">  -v, --verbose         Verbose output.</span></div></div></div><button type="button" aria-label="Copy code to clipboard" class="copyButton_2GIj">Copy</button></div></div><h2><a aria-hidden="true" tabindex="-1" class="anchor enhancedAnchor_prK2" id="genoml-discrete-supervised-tune"></a>genoml discrete supervised tune<a class="hash-link" href="#genoml-discrete-supervised-tune" title="Direct link to heading">#</a></h2><div class="mdxCodeBlock_1zKU"><div class="codeBlockContent_actS"><div tabindex="0" class="prism-code language-undefined codeBlock_tuNs thin-scrollbar"><div class="codeBlockLines_3uvA" style="color:#bfc7d5;background-color:#292d3e"><div class="token-line" style="color:#bfc7d5"><span class="token plain">usage: genoml discrete supervised tune [-h] [--prefix PREFIX]</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">                                       [--metric_tune {AUC,Balanced_Accuracy}]</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">                                       [--max_tune MAX_TUNE] [--n_cv N_CV]</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">                                       [-v]</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain" style="display:inline-block">
</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">optional arguments:</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">  -h, --help            show this help message and exit</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">  --prefix PREFIX       Prefix for your output build.</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">  --metric_tune {AUC,Balanced_Accuracy}</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">                        Using what metric of the best algorithm do you want to</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">                        tune on? [default: AUC].</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">  --max_tune MAX_TUNE   Max number of tuning iterations: (integer likely</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">                        greater than 10). This governs the length of tuning</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">                        process, run speed and the maximum number of possible</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">                        combinations of tuning parameters [default: 50].</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">  --n_cv N_CV           Number of cross validations: (integer likely greater</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">                        than 3). Here we set the number of cross-validation</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">                        runs for the algorithms [default: 5].</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">  -v, --verbose         Verbose output.</span></div></div></div><button type="button" aria-label="Copy code to clipboard" class="copyButton_2GIj">Copy</button></div></div><h2><a aria-hidden="true" tabindex="-1" class="anchor enhancedAnchor_prK2" id="genoml-continuous-supervised-tune"></a>genoml continuous supervised tune<a class="hash-link" href="#genoml-continuous-supervised-tune" title="Direct link to heading">#</a></h2><div class="mdxCodeBlock_1zKU"><div class="codeBlockContent_actS"><div tabindex="0" class="prism-code language-undefined codeBlock_tuNs thin-scrollbar"><div class="codeBlockLines_3uvA" style="color:#bfc7d5;background-color:#292d3e"><div class="token-line" style="color:#bfc7d5"><span class="token plain">usage: genoml continuous supervised tune [-h] [--prefix PREFIX]</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">                                         [--max_tune MAX_TUNE] [--n_cv N_CV]</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">                                         [-v]</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain" style="display:inline-block">
</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">optional arguments:</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">  -h, --help           show this help message and exit</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">  --prefix PREFIX      Prefix for your output build.</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">  --max_tune MAX_TUNE  Max number of tuning iterations: (integer likely</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">                       greater than 10). This governs the length of tuning</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">                       process, run speed and the maximum number of possible</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">                       combinations of tuning parameters [default: 50].</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">  --n_cv N_CV          Number of cross validations: (integer likely greater</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">                       than 3). Here we set the number of cross-validation</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">                       runs for the algorithms [default: 5].</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">  -v, --verbose        Verbose output.</span></div></div></div><button type="button" aria-label="Copy code to clipboard" class="copyButton_2GIj">Copy</button></div></div><h2><a aria-hidden="true" tabindex="-1" class="anchor enhancedAnchor_prK2" id="genoml-discrete-supervised-test"></a>genoml discrete supervised test<a class="hash-link" href="#genoml-discrete-supervised-test" title="Direct link to heading">#</a></h2><div class="mdxCodeBlock_1zKU"><div class="codeBlockContent_actS"><div tabindex="0" class="prism-code language-undefined codeBlock_tuNs thin-scrollbar"><div class="codeBlockLines_3uvA" style="color:#bfc7d5;background-color:#292d3e"><div class="token-line" style="color:#bfc7d5"><span class="token plain">usage: genoml discrete supervised test [-h] [--prefix PREFIX]</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">                                       [--test_prefix TEST_PREFIX]</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">                                       [--ref_model_prefix REF_MODEL_PREFIX]</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">                                       [-v]</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain" style="display:inline-block">
</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">optional arguments:</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">  -h, --help            show this help message and exit</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">  --prefix PREFIX       Prefix for your output build.</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">  --test_prefix TEST_PREFIX</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">                        Prefix for the dataset you would like to test against</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">                        your reference model. Remember, the model will not</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">                        function well if it does not include the same</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">                        features, and these features should be on the same</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">                        numeric scale, you can leave off the &#x27;.dataForML.h5&#x27;</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">                        suffix.</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">  --ref_model_prefix REF_MODEL_PREFIX</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">                        Prefix of your reference model file, you can leave off</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">                        the &#x27;.joblib&#x27; suffix.</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">  -v, --verbose         Verbose output.</span></div></div></div><button type="button" aria-label="Copy code to clipboard" class="copyButton_2GIj">Copy</button></div></div><h2><a aria-hidden="true" tabindex="-1" class="anchor enhancedAnchor_prK2" id="genoml-continuous-supervised-test"></a>genoml continuous supervised test<a class="hash-link" href="#genoml-continuous-supervised-test" title="Direct link to heading">#</a></h2><div class="mdxCodeBlock_1zKU"><div class="codeBlockContent_actS"><div tabindex="0" class="prism-code language-undefined codeBlock_tuNs thin-scrollbar"><div class="codeBlockLines_3uvA" style="color:#bfc7d5;background-color:#292d3e"><div class="token-line" style="color:#bfc7d5"><span class="token plain">usage: genoml continuous supervised test [-h] [--prefix PREFIX]</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">                                         [--test_prefix TEST_PREFIX]</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">                                         [--ref_model_prefix REF_MODEL_PREFIX]</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">                                         [-v]</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain" style="display:inline-block">
</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">optional arguments:</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">  -h, --help            show this help message and exit</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">  --prefix PREFIX       Prefix for your output build.</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">  --test_prefix TEST_PREFIX</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">                        Prefix for the dataset you would like to test against</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">                        your reference model. Remember, the model will not</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">                        function well if it does not include the same</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">                        features, and these features should be on the same</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">                        numeric scale, you can leave off the &#x27;.dataForML.h5&#x27;</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">                        suffix.</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">  --ref_model_prefix REF_MODEL_PREFIX</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">                        Prefix of your reference model file, you can leave off</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">                        the &#x27;.joblib&#x27; suffix.</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">  -v, --verbose         Verbose output.</span></div></div></div><button type="button" aria-label="Copy code to clipboard" class="copyButton_2GIj">Copy</button></div></div></div></article><div class="margin-vert--xl"><div class="row"><div class="col"><a href="https://github.com/genoml/genoml.github.io/edit/main/docs/reference/cli.md" target="_blank" rel="noreferrer noopener"><svg fill="currentColor" height="1.2em" width="1.2em" preserveAspectRatio="xMidYMid meet" role="img" viewBox="0 0 40 40" class="iconEdit_2LL7"><g><path d="m34.5 11.7l-3 3.1-6.3-6.3 3.1-3q0.5-0.5 1.2-0.5t1.1 0.5l3.9 3.9q0.5 0.4 0.5 1.1t-0.5 1.2z m-29.5 17.1l18.4-18.5 6.3 6.3-18.4 18.4h-6.3v-6.2z"></path></g></svg>Edit this page</a></div></div></div><div class="margin-vert--lg"><nav class="pagination-nav" aria-label="Blog list page navigation"><div class="pagination-nav__item"><a class="pagination-nav__link" href="/docs/use-cases/cohort-stratification"><div class="pagination-nav__sublabel">Previous</div><div class="pagination-nav__label"> Cohort stratification</div></a></div><div class="pagination-nav__item pagination-nav__item--next"></div></nav></div></div></div><div class="col col--3"><div class="tableOfContents_2xL- thin-scrollbar"><ul class="table-of-contents table-of-contents__left-border"><li><a href="#genoml---the-root-of-all-commands" class="table-of-contents__link">genoml - the root of all commands</a></li><li><a href="#genoml-harmonize" class="table-of-contents__link">genoml harmonize</a></li><li><a href="#genoml-discrete-supervised-munge" class="table-of-contents__link">genoml discrete supervised munge</a></li><li><a href="#genoml-continuous-supervised-munge" class="table-of-contents__link">genoml continuous supervised munge</a></li><li><a href="#genoml-discrete-supervised-train" class="table-of-contents__link">genoml discrete supervised train</a></li><li><a href="#genoml-continuous-supervised-train" class="table-of-contents__link">genoml continuous supervised train</a></li><li><a href="#genoml-discrete-supervised-tune" class="table-of-contents__link">genoml discrete supervised tune</a></li><li><a href="#genoml-continuous-supervised-tune" class="table-of-contents__link">genoml continuous supervised tune</a></li><li><a href="#genoml-discrete-supervised-test" class="table-of-contents__link">genoml discrete supervised test</a></li><li><a href="#genoml-continuous-supervised-test" class="table-of-contents__link">genoml continuous supervised test</a></li></ul></div></div></div></div></main></div></div><footer class="footer footer--dark"><div class="container"><div class="row footer__links"><div class="col footer__col"><h4 class="footer__title">Docs</h4><ul class="footer__items"><li class="footer__item"><a class="footer__link-item" href="/docs/about/what-is-genoml/">Getting Started</a></li><li class="footer__item"><a class="footer__link-item" href="/docs/about/contribute">Contribute</a></li></ul></div><div class="col footer__col"><h4 class="footer__title">Community</h4><ul class="footer__items"><li class="footer__item"><a href="https://stackoverflow.com/questions/tagged/genoml" target="_blank" rel="noopener noreferrer" class="footer__link-item">Stack Overflow</a></li><li class="footer__item"><a href="https://twitter.com/geno_ml" target="_blank" rel="noopener noreferrer" class="footer__link-item">Twitter</a></li><li class="footer__item"><a href="https://discordapp.com/invite/genoml" target="_blank" rel="noopener noreferrer" class="footer__link-item">Discord</a></li></ul></div><div class="col footer__col"><h4 class="footer__title">More</h4><ul class="footer__items"><li class="footer__item"><a class="footer__link-item" href="/blog">Blog</a></li><li class="footer__item"><a href="https://github.com/genoml/genoml" target="_blank" rel="noopener noreferrer" class="footer__link-item">GitHub</a></li></ul></div></div><div class="footer__bottom text--center"><div class="footer__copyright">Copyright  2021 Data Tecnica International, LLC.</div></div></div></footer></div>
<script src="/styles.9c4bafb9.js"></script>
<script src="/runtime~main.dc45b22d.js"></script>
<script src="/main.8d3b9ef6.js"></script>
<script src="/1.99cc7029.js"></script>
<script src="/2.9d8f953f.js"></script>
<script src="/37.2b52632f.js"></script>
<script src="/38.8fd46ecf.js"></script>
<script src="/935f2afb.5192d914.js"></script>
<script src="/17896441.058f9384.js"></script>
<script src="/91c76d4c.eb124e00.js"></script>
</body>
</html>